### DISCUSSION TITLE
Help with STM32 (Nucleo-G474RE) and ST7789 - 4-wire-SPI ?

### DISCUSSION THREAD

**Discussion by YePererva** [Bodmer/TFT_eSPI#2983] [Q&A - General] [answered]
Hello everyone!

I'm trying to launch the ST7789-based screen (2.0 IPS 240x320, module is based on 22 pin connector) with STM32G474RET6U (from Nucleo-G747 board) with following wiring:

| screen module header pin | Nucleo pin |
| - | - |
| GND | GND |
|VCC | +3.3V |
| SCL | PA_5 : SCK / D13 on Arduino header |
| SDA |  PA_7 : MOSI / D11 on Arduino header |
| RES | PC_7 : D9 on Arduino header |
| DC | PA_9 : D8 on Arduino header |
| CS | PB_6 : CS / D10 on Arduino header |
| BLK | either 3.3V or not connected due to build-in pull-up? |

This wiring works with the Adafruit TFT example library for ST7789 (so the screen module is working for sure). Is this reasonable, to assume that the same wiring would work with the TFT_eSPI library?

Attempt to upload an example from TFT_eSPI (Cellular_Automata) results in a black screen.
Please, guide me to what to adjust (and where)?   

Thank you!

**Comment by HaenZ33** [comment]
Hi YePererva,

did you setup the User_Setup.h file according to your pins and display?

**Reply by YePererva** [reply, marked as answer]
Yes... Already. Made it working using these settings:
```
#define USER_SETUP_INFO "User_Setup"

// Define STM32 to invoke optimised processor support (only for STM32)
#define STM32

// Didn't find any difference between drivers
// #define ST7789_DRIVER
#define ST7789_2_DRIVER

// ST7789 240 x 320 portrait
#define TFT_WIDTH  240 
#define TFT_HEIGHT 320

// The TFT can be connected to SPI port 1 or 2
#define TFT_SPI_PORT 1 // SPI port 1 maximum clock rate is 55MHz
#define TFT_MOSI D11
#define TFT_MISO D12
#define TFT_SCLK D13
#define TFT_CS   D10 // Chip select control pin to TFT CS
#define TFT_DC   D8 // Data Command control pin to TFT DC (may be labelled RS = Register Select)
#define TFT_RST  D9 // Reset pin to TFT RST (or RESET)

#define SPI_FREQUENCY  55000000 // STM32 SPI1 only (SPI2 maximum is 27MHz)
```
